agent/test-requirements.txt

```diff
@@ -5,5 +5,5 @@ mock>=3.0.5
 pytest-cov>=2.7.1
 pytest>=4.6.3, < 5
 pytest-helpers-namespace==2019.1.8
-pytest-mock>=1.10.4
+pytest-mock>=1.10.4, <3.3
```

Comment:
I agree that it would be most excellent to document all of our restrictions; however, as soon as plausible, I fully intend to remove the upper limit on the requirement that you've cited here (I would have done it in this PR, but I expect it to be a non-trivial impact and effort, so I'm forcing myself to do it in another PR), so, with any luck, your request will be moot in this case.  (Although, "for safety sake", I suppose I could annotate this one anyway....)

With respect to formatting, [PEP 508](https://www.python.org/dev/peps/pep-0508/#examples) seems to agree with you, although, tragically, `pip freeze` does not.  😬  Personally, I don't like the extra space between the qualifier and the version, but, since I've a goal to remove the upper limits, that's probably not a big problem; and, with the upper limit gone the extra space will look "right" (to me).  (But, I'm not going to reformat the whole file....)